src/GraphControl/Control/Grapher.cpp

```diff
@@ -433,6 +436,108 @@ namespace GraphControl
         }
     }
 
+    shared_ptr<IGraph> Grapher::GetGraph(GraphControl::Equation ^ equation)
+    {
+        std::shared_ptr<Graphing::IGraph> graph = m_solver->CreateGrapher();
+
+        wstringstream ss{};
+        ss << L"<math xmlns=\"http://www.w3.org/1998/Math/MathML\"><mrow><mi>show2d</mi><mfenced separators=\"\">";
+        ss << equation->GetRequest();
+        ss << L"</mfenced></mrow></math>";
+
+        wstring request = ss.str();
+        unique_ptr<IExpression> graphExpression;
+        if (graphExpression = m_solver->ParseInput(request))
+        {
+            if (graph->TryInitialize(graphExpression.get()))
+            {
+                return graph;
+            }
+        }
+
+        return nullptr;
+    }
+
+    void Grapher::UpdateKeyGraphFeatures()
+    {
+        auto equations = GetValidEquations();
+        for (auto equation : GetValidEquations())
+        {
+            if (auto graph = GetGraph(equation))
+            {
+                if (auto analyzer = graph->GetAnalyzer())
+                {
+                    bool result;
+                    analyzer->CanFunctionAnalysisBePerformed(result);
+                    if (result)
+                    {
+                        m_solver->FormatOptions().SetFormatType(FormatType::MathML);
```

Comment:
This is the same solver we use for the main graph right? We should just set this on initialization instead of in here.

Reply:
hmm...moving it unset the value. I'll dig in.